### PR TITLE
Service catalog tests

### PR DESCRIPTION
**Pull request by kornys**

### CODE REVIEW

File: systemtests/src/test/java/io/enmasse/systemtest/common/catalog/ServiceCatalogWebTest.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+//        ConsoleWebPage consoleWebPage = ocPage.clickOnDashboard();
+//        consoleWebPage
+        waitForAddressSpaceReady(brokeredSpace);
+        reloadAddressSpaceEndpoints(brokeredSpace);
+        //this is important due to update of endpoint which are usually updated
+
+        ConsoleWebPage consolePage = new ConsoleWebPage(selenium, getConsoleRoute(brokeredSpace),
+                addressApiClient, brokeredSpace, developer);
+        consolePage.openWebConsolePage(true);
+        fail("korny za to nemuze");
```
Comment (zkraus) [inline systemtests/src/test/java/io/enmasse/systemtest/common/catalog/ServiceCatalogWebTest.java]: :+1:

File: systemtests/src/main/java/io/enmasse/systemtest/TestUtils.java [hunk 2]
```diff
@@ -305,6 +307,22 @@ public static boolean isAddressSpaceReady(JsonObject addressSpace) {
         return isReady;
     }
 
+    /**
+     * Check if state and description values equals required values
+     *
+     * @param status answer from service broker
```
Comment (tkratky) [inline systemtests/src/main/java/io/enmasse/systemtest/TestUtils.java]: add better description

File: systemtests/src/main/java/io/enmasse/systemtest/selenium/page/OpenshiftWebPage.java [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+
+    private WebElement getProjectListItem(String name) {
+        selenium.takeScreenShot();
+        List<WebElement> projects = selenium.getDriver().findElements(By.className("list-group-item"));
+        for (WebElement el : projects) {
+            if (el.findElement(By.tagName("span")).getText().equals(name)) {
+                return el;
+            }
+        }
+        return null;
```
Comment (tkratky) [inline systemtests/src/main/java/io/enmasse/systemtest/selenium/page/OpenshiftWebPage.java]: Exception should be better

File: systemtests/src/main/java/io/enmasse/systemtest/selenium/page/OpenshiftWebPage.java [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+                selenium.clickOnItem(element);
+            }
+        }
+    }
+
+    private void selectProjectInWizard(String projectName) throws Exception {
+        clickOnAddToProjectDropdown();
+        WebElement project = getItemFromAddToProjectDropDown(projectName);
+        if (project != null) {
+            log.info("Project is present address space will be added into them: {}", projectName);
```
Comment (tkratky) [inline systemtests/src/main/java/io/enmasse/systemtest/selenium/page/OpenshiftWebPage.java]: them->it ?

File: systemtests/src/main/java/io/enmasse/systemtest/selenium/page/OpenshiftWebPage.java [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+            allowExternalAccess();
+        }
+        if (receiveAddress != null) {
+            fillReceiveAddresses(receiveAddress);
+        }
+        if (sendAddresses != null) {
+            fillSendAddresses(sendAddresses);
+        }
+        next();
+        Thread.sleep(2000);
```
Comment (tkratky) [inline systemtests/src/main/java/io/enmasse/systemtest/selenium/page/OpenshiftWebPage.java]: is it required?

File: systemtests/src/main/java/io/enmasse/systemtest/selenium/page/OpenshiftWebPage.java [hunk 6]
```diff
@@ -0,0 +1,383 @@
...
+        ProvisionedServiceItem serviceItem = getProvisionedServiceItem();
+        serviceItem.collapseServiceItem();
+        selenium.clickOnItem(serviceItem.getRedirectConsoleButton());
+        Set<String> tabHandles = selenium.getDriver().getWindowHandles();
+        selenium.getDriver().switchTo().window(tabHandles.toArray()[tabHandles.size() - 1].toString());
+        return new ConsoleWebPage(selenium, addressApiClient, addressSpace);
+    }
+
+    @Override
+    public void checkReachableWebPage() {
```
Comment (tkratky) [inline systemtests/src/main/java/io/enmasse/systemtest/selenium/page/OpenshiftWebPage.java]: create low priority issue for implementation those methods (for all web pages)

File: systemtests/src/test/java/io/enmasse/systemtest/common/catalog/ServiceCatalogApiTest.java [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+                try {
+                    osbApiClient.deprovisionInstance(space, developer.getUsername(), id);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            });
+            instances.clear();
+        } else {
+            log.warn("Remove service instances in tear down - SKIPPED!");
+        }
```
Comment (tkratky) [inline systemtests/src/test/java/io/enmasse/systemtest/common/catalog/ServiceCatalogApiTest.java]: if (!environment.useMinikube()) {
stop osbapi client

File: systemtests/src/test/java/io/enmasse/systemtest/common/catalog/ServiceCatalogApiTest.java [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+    private void provideAndCreateBinding(AddressSpace addressSpace, String wildcardMark) throws Exception {
+        ServiceInstance provInstance = createServiceInstance(addressSpace, developer.getUsername());
+        HashMap<String, String> bindResources = new HashMap<>();
+        bindResources.put("sendAddresses", String.format("queue.%s", wildcardMark));
+        bindResources.put("receiveAddresses", String.format("queue.%s", wildcardMark));
+        bindResources.put("consoleAccess", "true");
+        bindResources.put("consoleAdmin", "false");
+        bindResources.put("externalAccess", "false");
+        String bindingId = generateBinding(addressSpace, developer.getUsername(), provInstance.getInstanceId(), bindResources);
+
```
Comment (tkratky) [inline systemtests/src/test/java/io/enmasse/systemtest/common/catalog/ServiceCatalogApiTest.java]: add TODO comment into code for testing of bindings (send/receive/console....)

File: systemtests/src/test/java/io/enmasse/systemtest/common/catalog/ServiceCatalogApiTest.java [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+    @DisabledIfEnvironmentVariable(named = useMinikubeEnv, matches = "true")
+    void testMissingRequiredParameter() throws Exception {
+        AddressSpace addressSpaceViaOSBAPI = new AddressSpace(null, AddressSpaceType.BROKERED);
+        createServiceInstance(addressSpaceViaOSBAPI, developer.getUsername());
+
+        //!TODO there are missing steps what should happen when required parameter is missing
+    }
+
+    @Test
+    @DisabledIfEnvironmentVariable(named = useMinikubeEnv, matches = "true")
```
Comment (tkratky) [inline systemtests/src/test/java/io/enmasse/systemtest/common/catalog/ServiceCatalogApiTest.java]: add @ Disable("missing button on console login web page")

File: systemtests/src/test/java/io/enmasse/systemtest/common/catalog/ServiceCatalogWebTest.java [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+        consolePage.openWebConsolePage(true);
+
+        //do simple send/receive
+        Destination queue = Destination.queue("test-queue", "brokered-queue");
+        consolePage.createAddressWebConsole(queue, false, true);
+
+        AmqpClient client = amqpClientFactory.createQueueClient(brokeredSpace);
+        client.getConnectOptions().setCredentials(developer);
+        QueueTest.runQueueTest(client, queue);
+
```
Comment (tkratky) [inline systemtests/src/test/java/io/enmasse/systemtest/common/catalog/ServiceCatalogWebTest.java]: remove

File: systemtests/src/main/java/io/enmasse/systemtest/TestUtils.java (hunk 2 above)
Comment (tkratky) [reply systemtests/src/main/java/io/enmasse/systemtest/TestUtils.java]: done

File: systemtests/src/main/java/io/enmasse/systemtest/selenium/page/OpenshiftWebPage.java (hunk 6 above)
Comment (tkratky) [reply systemtests/src/main/java/io/enmasse/systemtest/selenium/page/OpenshiftWebPage.java]: new issue: https://github.com/EnMasseProject/enmasse/issues/1230

File: systemtests/src/test/java/io/enmasse/systemtest/common/catalog/ServiceCatalogApiTest.java (hunk 7 above)
Comment (tkratky) [reply systemtests/src/test/java/io/enmasse/systemtest/common/catalog/ServiceCatalogApiTest.java]: done

File: systemtests/src/test/java/io/enmasse/systemtest/common/catalog/ServiceCatalogApiTest.java (hunk 8 above)
Comment (tkratky) [reply systemtests/src/test/java/io/enmasse/systemtest/common/catalog/ServiceCatalogApiTest.java]: done

File: systemtests/src/main/java/io/enmasse/systemtest/selenium/page/OpenshiftWebPage.java (hunk 3 above)
Comment (kornys) [reply systemtests/src/main/java/io/enmasse/systemtest/selenium/page/OpenshiftWebPage.java]: there is ok with null, parent method handle null state exception.

File: systemtests/src/test/java/io/enmasse/systemtest/common/catalog/ServiceCatalogApiTest.java (hunk 9 above)
Comment (tkratky) [reply systemtests/src/test/java/io/enmasse/systemtest/common/catalog/ServiceCatalogApiTest.java]: done

Comment (tkratky) [reply systemtests/src/test/java/io/enmasse/systemtest/common/catalog/ServiceCatalogApiTest.java]: done

**Comment by kornys** [comment]
@tkratky branch is re-based on actual master (master contains fixes for binding). I will wait for results in CI and merge it with master.

**Pull request merged**